### PR TITLE
!deploy and add Chocopin feature

### PR DESCRIPTION
**Pull request by Outek**
Please review the new Function and resources for Chocolatey.
I made also some code formatting changes, i hope you are ok with this.

### CODE REVIEW

**Comment by gaelcolas** [comment]
Hey, thanks a lot for that!
Gonna need to find some time to look at this, and that probably won't be this weekend. Do give me a nudge if you don't get feedback in a week.

**Comment by Outek** [comment]
"Give Me A Ping, Vasily. One Ping Only, Please."

**Comment by gaelcolas** [comment]
Note for later, please don't mix formatting change with feature change. It makes reviewing so much harder :(
Going through the changes, slowly...

**Comment by gaelcolas** [comment]
Wow, great, extensive work! thanks.
I'm getting there.

**Comment by Outek** [comment]
> Note for later, please don't mix formatting change with feature change. It makes reviewing so much harder `:(`
> Going through the changes, slowly...

I'm terribly sorry for that mess....

File: Chocolatey/public/Get-ChocolateySetting.ps1 [hunk 1]
```diff
@@ -40,16 +40,16 @@ function Get-ChocolateySetting {
...
     Process {
         if (!$ChocoXml) {
-            Throw "Error with Chocolatey config"
+            Throw "Error with Chocolatey config."
         }
 
         foreach ($Name in $Setting) {
             if ($Name -ne '*') {
-                Write-Verbose ('Searching for Setting named ${0}' -f [Security.SecurityElement]::Escape($Name))
+                Write-Verbose ("Searching for Setting named {0}" -f [Security.SecurityElement]::Escape($Name))
```
Comment (gaelcolas) [inline Chocolatey/public/Get-ChocolateySetting.ps1]: This change is not necessary.

File: Chocolatey/tests/Unit/Private/Repair-PowerShellOutputRedirectionBug.tests.ps1 [hunk 2]
```diff
@@ -4,7 +4,7 @@ InModuleScope Chocolatey {
         Context 'Default' {
 
             It 'Should apply the fix silently' {
-                { Repair-PowerShellOutputRedirectionBug } | Should not throw
+                { Repair-PowerShellOutputRedirectionBug } | Should not Throw
```
Comment (gaelcolas) [inline Chocolatey/tests/Unit/Private/Repair-PowerShellOutputRedirectionBug.tests.ps1]: This one should be: `| Should -not -throw` iirc (for the latest version of Pester)

File: Chocolatey/tests/Unit/Public/Install-ChocolateySoftware.tests.ps1 [hunk 3]
```diff
@@ -20,18 +20,18 @@ InModuleScope Chocolatey {
 
             It 'Ensure Get-RemoteString is called when install from Feed without version' {
                 $null = Install-ChocolateySoftware
-                { Assert-MockCalled Get-RemoteString } | Should Not Throw
+                { Assert-MockCalled Get-RemoteString } | Should not Throw
             }
```
Comment (gaelcolas) [inline Chocolatey/tests/Unit/Public/Install-ChocolateySoftware.tests.ps1]: Again: `| Should -not -Throw`

**Pull request merged**